lib/bsdlib/nrf91_sockets.c

```diff
@@ -827,6 +844,17 @@ static int nrf91_socket_offload_getaddrinfo(const char *node,
 	}
 	int retval = nrf_getaddrinfo(node, service, nrf_hints_ptr, &nrf_res);
 
+	if (retval != 0) {
+		error = nrf_to_z_error_code(retval);
+		if (error == DNS_EAI_SYSTEM) {
+			errno = retval;
+                        retval = DNS_EAI_SYSTEM;
+		}
+		else {
+			retval = error;
+		}
```

Comment:
I think we should return here. Otherwise, we'll call `freeaddrinfo` on unallocated `nrf_res` .

Reply:
Return added